packages/mattermost-redux/src/reducers/entities/threads/counts.ts

```diff
@@ -142,8 +147,26 @@ export function countsIncludingDirectReducer(state: ThreadsState['counts'] = {},
     switch (action.type) {
     case ThreadTypes.ALL_TEAM_THREADS_READ:
         return handleAllTeamThreadsRead(state, action);
-    case ThreadTypes.READ_CHANGED_THREAD:
-        return handleReadChangedThread(state, action);
+    case ThreadTypes.READ_CHANGED_THREAD: {
+        const {teamId, channelType} = action.data;
+        if (teamId === '' || isDmGmChannel(channelType)) {
+            const teamIds = new Set(Object.keys(state));
+
+            // if the case of dm/gm make sure we add counts for all teams
+            if (teamId !== '') {
+                teamIds.add(teamId);
+            }
+
+            let newState = {...state};
+            teamIds.forEach((teamId) => {
```

Comment:
It might be nice to give this a different name to avoid shadowing `teamId`. I'm actually surprised ESLint doesn't complain about that